apps/src/code-studio/components/ExportDialog/Dialog.jsx

```diff
@@ -446,13 +466,22 @@ class ExportDialog extends React.Component {
         this.setState({screen: 'icon'});
         break;
       case 'icon':
-        this.setState({screen: 'publish'});
+        this.setState({
+          screen: platform === 'android' ? 'publishAndroid' : 'publishIOS'
+        });
         break;
-      case 'publish':
+      case 'publishAndroid':
         this.generateApkAsNeeded();
         this.setState({screen: 'generating'});
         break;
+      case 'publishIOS':
+        this.publishExpoExport();
```

Comment:
Perhaps should be named `generateIpaAsNeeded()`

Reply:
This `publishExpoExport()` step is used by both iOS and Android.
For Android, we call `generateApkAsNeeded()`, which calls `publishAndGenerateApk()`, which does a 2 step process: first `publishExpoExport()`, then `expoGenerateApk()`
For iOS, we just do `publishExpoExport()`, and then we transition to Expo.io to continue.

Reply:
👍